packages/fluentui/react-northstar/src/components/Datepicker/Datepicker.tsx

```diff
@@ -322,55 +325,51 @@ export const Datepicker: ComponentWithAs<'div', DatepickerProps> &
     <Button icon={<CalendarIcon />} title={props.openCalendarTitle} iconOnly disabled={props.disabled} />
   );
 
-  const element = (
-    <Ref innerRef={datepickerRef}>
-      {getA11yProps.unstable_wrapWithFocusZone(
-        <ElementType
-          {...getA11yProps('root', {
-            className: classes.root,
-            ...unhandledProps,
-          })}
-        >
-          {!props.buttonOnly &&
-            createShorthand(Input, input, {
-              defaultProps: () =>
-                getA11yProps('input', {
-                  placeholder: props.inputPlaceholder,
-                  disabled: props.disabled,
-                  error: !!error,
-                  value: formattedDate,
-                  readOnly: !allowManualInput,
-                  required: props.required,
-                  ref: inputRef,
-                  'aria-label': formatRestrictedInput(restrictedDatesOptions, dateFormatting),
-                }),
-              overrideProps: overrideInputProps,
-            })}
-          {createShorthand(Popup, popup, {
-            defaultProps: () => ({
-              open: openState && !props.disabled,
-              content: calendarElement,
-              trapFocus: {
-                disableFirstFocus: true,
-              },
-              trigger: triggerButtonElement,
-              target: !props.buttonOnly ? inputRef.current : null,
-              position: 'below' as const,
-              align: 'start' as const,
+  const element = getA11yProps.unstable_wrapWithFocusZone(
+    <ElementType
+      {...getA11yProps('root', {
+        className: classes.root,
+        ...unhandledProps,
+      })}
+    >
+      {!props.buttonOnly &&
+        createShorthand(Input, input, {
+          defaultProps: () =>
+            getA11yProps('input', {
+              placeholder: props.inputPlaceholder,
+              disabled: props.disabled,
+              error: !!error,
+              value: formattedDate,
+              readOnly: !allowManualInput,
+              required: props.required,
+              ref: inputRef,
```

Comment:
This should be in `overrideProps` and perform merge as if I will pass: `<Datepicker input={{ ref }} />` it will be broken

Reply:
Curious: 
In what scenarios might the user want to override the inputRef? 

Except for the overrideProps section within Input, how would we get access to the user's entered value (context: Popup.onOpenChange; Popup.target)?

Reply:
> In what scenarios might the user want to override the inputRef?

User does not want to override a ref, but wants to get a `ref` and then it will cause weird bugs:
https://github.com/microsoft/fluentui/issues/13661
https://github.com/microsoft/fluentui/pull/13711

Reply:
Updated.